creation/web_base/cvmfs_helper_funcs.sh

```diff
@@ -30,476 +30,522 @@
-	#	-> status of unpriv userns (unavailable, disabled, enabled,
-	#	error) to stdout
-
-	# make sure that perform_system_check has run
-	[[ -z "${GWMS_SYSTEM_CHECK}" ]] && perform_system_check
-
-	# determine whether unprivileged user namespaces are supported and/or enabled...
-	if [[ "${GWMS_IS_UNPRIV_USERNS_ENABLED}" -eq 0 ]]; then
-		# unprivileged user namespaces is enabled
-		if [[ "${GWMS_IS_UNPRIV_USERNS_SUPPORTED}" -eq 0 ]]; then
-			# unprivileged user namespaces is supported
-			loginfo "Unprivileged user namespaces supported and enabled"
-			echo enabled
-		else
-			# unprivileged user namespaces is not supported
-			logerror "Inconsistent system configuration: unprivileged userns is enabled but not supported"
-        		echo error
-		fi
-		true
-	else
-		# unprivileged user namespaces is disabled
-		if [[ "${GWMS_IS_UNPRIV_USERNS_SUPPORTED}" -eq 0 ]]; then
-			# unprivileged user namespaces is supported
-			logwarn "Unprivileged user namespaces disabled: can be enabled by the root user via sysctl"
-			echo disabled
-		else
-			# unprivileged user namespaces is not supported
-			logwarn "Unprivileged user namespaces disabled and unsupported: can be supported/enabled only after a system upgrade"
-			echo unavailable
-		fi
-		false
-	fi
+get_mount_point() {
+    # TODO: Verify the findmnt ... will always find the correct CVMFS mount
+    mount_point=$(findmnt -t fuse -S /dev/fuse | tail -n 1 | cut -d ' ' -f 1 )
+    if [[ -n "$mount_point" && "$mount_point" != TARGET* ]]; then
+        mount_point=$(dirname "$mount_point")
+        if [[ -n "$mount_point" && "$mount_point" != /cvmfs ]]; then
+            CVMFS_MOUNT_DIR="$mount_point"
+            export CVMFS_MOUNT_DIR=$mount_point
+            gconfig_add CVMFS_MOUNT_DIR "$mount_point"
+        fi
+    fi
+}
+
 
+has_unpriv_userns() {
+    # DESCRIPTION: This function checks the status of unprivileged user
+    # namespaces being supported and enabled on the worker node. Depending
+    #
+    # INPUT(S): None
+    # RETURN(S):
+    #	-> true (0) if unpriv userns can be used (supported and enabled),
+    #	false otherwise
+    #	-> status of unpriv userns (unavailable, disabled, enabled,
+    #	error) to stdout
+
+    # make sure that perform_system_check has run
+    [[ -z "${GWMS_SYSTEM_CHECK}" ]] && perform_system_check
+
+    # determine whether unprivileged user namespaces are supported and/or enabled...
+    if [[ "${GWMS_IS_UNPRIV_USERNS_ENABLED}" -eq 0 ]]; then
+        # unprivileged user namespaces is enabled
+        if [[ "${GWMS_IS_UNPRIV_USERNS_SUPPORTED}" -eq 0 ]]; then
+            # unprivileged user namespaces is supported
+            loginfo "Unprivileged user namespaces supported and enabled"
+            echo enabled
+        else
+            # unprivileged user namespaces is not supported
+            logerror "Inconsistent system configuration: unprivileged userns is enabled but not supported"
+                echo error
+        fi
+        true
```

Comment:
That's a great question. This function `has_unpriv_userns` is a helper method that is invoked by another function `perform_cvmfs_mount` (defined later in this script) that uses the value echoed from this function to determine whether cvmfs mount can be performed or not.

Having said that, I don't exactly remember why I added the `true` and `false` statements like the way I did. I'm guessing it might have been because I wanted to capture the exit status of each of the four conditional checks in case that had to be used later. But since the exit status is not at all used but only the value that was echoed, we could remove the true/false statements.